src/transformers/integrations.py

```diff
@@ -1116,6 +1127,20 @@ def _log_hyper_param_search_parameters(self, state):
         if state and hasattr(state, 'trial_params') and state.trial_params is not None:
             self._metadata_namespace['trial_params'] = state.trial_params
 
+    def _log_model_checkpoint(self, path: str):
+        self._metadata_namespace[self._target_checkpoints_namespace].upload_files(path)
+
+        if self._should_clean_recently_uploaded_checkpoint and self._recent_checkpoint_path is not None:
+            self._metadata_namespace[self._target_checkpoints_namespace].delete_files(self._recent_checkpoint_path)
+
+        self._recent_checkpoint_path = path
+
+    def on_init_end(self, args, state, control, **kwargs):
+        if self._log_checkpoints == 'best':
+            # TODO: NPT-12189 - Update the assertion about required arguments
+            assert args.load_best_model_at_end, "To use best model checkpoint saving load_best_model_at_end needs to " \
```

Comment:
Shouldn't we throw some internal exception instead of simply assertion?
I've been quite surprised when noticing in E2E that bad configuration raises assertion exception.

Reply:
What do you think about `ValueError`?

Reply:
ok

Reply:
Done